pkg/runtime/runtime.go

```diff
@@ -2602,7 +2626,35 @@ func (a *DaprRuntime) processComponentAndDependents(comp componentsV1alpha1.Comp
 	}
 
 	go func() {
-		ch <- a.doProcessOneComponent(compCategory, comp)
+		for {
+			log.Debugf("Processing component: %s", comp.LogName())
+			found, err := a.doProcessOneComponent(compCategory, comp)
+
+			if err != nil || found {
+				ch <- err
+				return
+			}
+
+			// A component not being found means one of two things:
+			//
+			// - The component is pluggable but hasn't yet been loaded
+			// - A mis-configuration
+			//
+			// Assuming the former, Dapr will wait a short interval (to allow the component to load), scan for pluggable components, and try again.
+			// If the latter, eventually the init timeout will be reached and Dapr will bail.
+
+			log.Debugf("Component type '%s' was not found", comp.Spec.Type)
+
+			if a.hasInitializedPluggableComponents {
+				log.Debugf("Waiting %s for pluggable components to load", defaultPluggableComponentWaitInterval.String())
```

Comment:
Some thoughts:

1. The default init timeout is 5s, and if it takes even just 2s to launch the container, then its init time is greatly reduced. Consider having a separate timeout for waiting for the component from the init timeout?
2. This log should be level `Info`, BUT you should only print it once and not every time you make an attempt to check if the component is loaded, to avoid putting too much noise in logs
3. Consider making the interval an incremental backoff (see how we use the backoff library)

Reply:
The startup time for a component should, generally speaking, be close to constant so I would *think* it ok for a user to consider that in any adjustment to the `initTimeout`.  In other words, I would expect any real initialization time to drown out startup time for the purposes of optimizing a timeout.

Reply:
My concern is that you're assuming users actually read the docs :)

The problem is that in this case the initialization has 2 ways to go:

- if the container is already up by the time the runtime is trying to init the component, then the constant init time will kick in
- On the other hand, if the container is not up yet, this will take longer

Unless I'm misunderstanding?